CMakeLists.txt

```diff
@@ -1,69 +1,68 @@
-cmake_minimum_required(VERSION 2.8.3)
-project(opw_kinematics)
-
-add_compile_options(-std=c++11 -Wall -Wextra)
-
-find_package(catkin REQUIRED)
+cmake_minimum_required(VERSION 3.8.0)
+project(opw_kinematics VERSION 0.1.0 LANGUAGES CXX)
 
 find_package(Eigen3 REQUIRED)
-if(NOT EIGEN3_INCLUDE_DIRS)
-  set(EIGEN3_INCLUDE_DIRS ${EIGEN3_INCLUDE_DIR})
-endif()
-
-catkin_package(
-  INCLUDE_DIRS include
-  DEPENDS EIGEN3
-)
-
-###########
-## Build ##
-###########
-
-include_directories(
-  include
-  ${catkin_INCLUDE_DIRS}
-  ${EIGEN3_INCLUDE_DIRS}
-)
 
+# Create interface target
+add_library(${PROJECT_NAME} INTERFACE)
+target_link_libraries(${PROJECT_NAME} INTERFACE)
+target_compile_options(${PROJECT_NAME} INTERFACE -Wall -Wextra)
+target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_11)
+target_include_directories(${PROJECT_NAME} INTERFACE
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+    "$<INSTALL_INTERFACE:include>")
+target_include_directories(${PROJECT_NAME} SYSTEM INTERFACE
+    $<TARGET_PROPERTY:Eigen3::Eigen,INTERFACE_INCLUDE_DIRECTORIES>)
```

Comment:
Next problem: this doesn't work with Eigen `< 3.3`, which is what Xenial ships with (see [this SO post](https://stackoverflow.com/questions/53991812)).